src/js/JournalContainer.js

```diff
@@ -46,19 +66,42 @@ const JournalTextEntry = () => {
   };
 
   const saveEntry = () => {
-    let timestamp = Date.now();
+    let entries = [...journalEntries];
+    let timestamp = Date(Date.now());
 
     const journalEntry = {
+      journal_id: id == undefined ? uuidv4() : findJournalEntry(id).journal_id,
+      prompt: document.getElementsByClassName('prompt')[0].innerHTML,
       title: title,
       content: value,
-      date: timestamp
+      date: timestamp,
+      type: id == undefined ? journalType : findJournalEntry(id).type
+    };
+
+    const spliceIndex = () => {
+      entries.splice(entries.indexOf(id), 1);
```

Comment:
The save function for previous entries is replacing the last entry and not deleting the specific entry. This is because it can’t find the index properly and it always returns as -1. This code should find the id: 
```suggestion
      const index = entries.findIndex((x) => x.journal_id === id);
```

Reply:
Okay, I committed this change. TBH I wasn't really sure how to find the index so thanks for correcting it.